### PR TITLE
Keep logs of Postgres clients tests

### PR DESCRIPTION
**Pull request by bayandin**
Should help to debug problems like this https://github.com/neondatabase/neon/runs/7799415466?check_suite_focus=true#step:7:114

### CODE REVIEW

File: .github/workflows/pg_clients.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name: Test Postgres client libraries
 
 on:
+  push:
```
Comment (SomeoneToIgnore) [inline .github/workflows/pg_clients.yml:4]: Let's not forget to revert this.

**Review by SomeoneToIgnore — APPROVED** [review]
Is there a way to see how the log output changed now?
A bit odd to see that the tests that use these changes were not run 🙂

**Comment by bayandin** [comment]
> A bit odd to see that the tests that use these changes were not run 🙂

[They were!](https://github.com/neondatabase/neon/runs/7804240439?check_suite_focus=true) But they didn't fail 🤷‍♂️ 

> Is there a way to see how the log output changed now? 

Yep, here is an example from one of the previous runs with failed test (still need to download an artifact to get logs): https://github.com/neondatabase/neon/actions/runs/2846006126

**Comment by SomeoneToIgnore** [comment]
I've checked csharp logs and that looks nice.
One note though, that there's plenty of logs with `1` in it, not sure if that's worth writing or collecting and looks like some mistake 🙂 

But I've noted the java test error, so it's much clearer now, thank you.

File: test_runner/fixtures/utils.py [hunk 2]
```diff
@@ -32,10 +32,16 @@ def subprocess_capture(capture_dir: str, cmd: List[str], **kwargs: Any) -> str:
...
     stderr_filename = basepath + '.stderr'
 
-    with open(stdout_filename, 'w') as stdout_f:
-        with open(stderr_filename, 'w') as stderr_f:
-            log.info('(capturing output to "{}.stdout")'.format(base))
-            subprocess.run(cmd, **kwargs, stdout=stdout_f, stderr=stderr_f)
+    try:
+        with open(stdout_filename, 'w') as stdout_f:
+            with open(stderr_filename, 'w') as stderr_f:
+                log.info('(capturing output to "{}.stdout")'.format(base))
```
[the remark this answers is not in the document]
Comment (bayandin) [reply test_runner/fixtures/utils.py:38]: We include both `stderr` and `stdout`, I'll update the misleading message

**Comment by bayandin** [comment]
> One note though, that there's plenty of logs with 1 in it, not sure if that's worth writing or collecting and looks like some mistake 🙂

It's actually the output which we check in the test (we call `SELECT 1` using different libraries). I will keep it for now 😄

**Pull request merged**